src/python/pants/engine/process.py

```diff
@@ -50,6 +51,31 @@ class ProcessCacheScope(Enum):
     PER_SESSION = "per_session"
 
 
+@dataclass(frozen=True)
+class ProcessConcurrency:
```

Comment:
I changed this type from the first draft you looked at @benjyw Its a single dataclass in python now, which I think is a little cleaner